src/GafferScene/PrimitiveSampler.cpp

```diff
@@ -0,0 +1,385 @@
+}
+
+const Gaffer::StringPlug *PrimitiveSampler::primitiveVariablesPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 2 );
+}
+
+Gaffer::StringPlug *PrimitiveSampler::prefixPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 3 );
+}
+
+const Gaffer::StringPlug *PrimitiveSampler::prefixPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 3 );
+}
+
+Gaffer::StringPlug *PrimitiveSampler::statusPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 4 );
+}
+
+const Gaffer::StringPlug *PrimitiveSampler::statusPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 4 );
+}
+
+bool PrimitiveSampler::affectsProcessedObject( const Gaffer::Plug *input ) const
+{
+	return
+		Deformer::affectsProcessedObject( input ) ||
+		input == sourcePlug() ||
+		input == sourceLocationPlug() ||
+		input == primitiveVariablesPlug() ||
+		input == prefixPlug() ||
+		input == statusPlug() ||
+		input == sourcePlug()->objectPlug() ||
+		input == inPlug()->transformPlug() ||
+		input == sourcePlug()->transformPlug() ||
+		affectsSamplingFunction( input )
+	;
+}
+
+void PrimitiveSampler::hashProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	PrimitiveSampler::Deformer::hashProcessedObject( path, context, h );
+
+	const string sourceLocation = sourceLocationPlug()->getValue();
+	const string primitiveVariables = primitiveVariablesPlug()->getValue();
+	const string status = statusPlug()->getValue();
+	if( sourceLocation.empty() || ( primitiveVariables.empty() && status.empty() ) )
+	{
+		return;
+	}
+
+	ScenePlug::ScenePath sourcePath;
+	ScenePlug::stringToPath( sourceLocation, sourcePath );
+	if( !SceneAlgo::exists( sourcePlug(), sourcePath ) )
+	{
+		return;
+	}
+
+	h.append( sourcePlug()->objectHash( sourcePath ) );
+	h.append( primitiveVariables );
+	prefixPlug()->hash( h );
+	h.append( status );
+	h.append( inPlug()->fullTransformHash( path ) );
+	h.append( sourcePlug()->fullTransformHash( sourcePath ) );
+	hashSamplingFunction( h );
+}
+
+IECore::ConstObjectPtr PrimitiveSampler::computeProcessedObject( const ScenePath &path, const Gaffer::Context *context, const IECore::Object *inputObject ) const
+{
+	const Primitive *inputPrimitive = runTimeCast<const Primitive>( inputObject );
+	if( !inputPrimitive )
+	{
+		return inputObject;
+	}
+
+	const string sourceLocation = sourceLocationPlug()->getValue();
+	const string primitiveVariables = primitiveVariablesPlug()->getValue();
+	const string status = statusPlug()->getValue();
+	if( sourceLocation.empty() || ( primitiveVariables.empty() && status.empty() ) )
+	{
+		return inputObject;
+	}
+
+	PrimitiveVariable::Interpolation outputInterpolation = PrimitiveVariable::Invalid;
+	SamplingFunction samplingFunction = computeSamplingFunction( inputPrimitive, outputInterpolation );
+	if( !samplingFunction || outputInterpolation == PrimitiveVariable::Invalid )
+	{
+		return inputObject;
+	}
```

Comment:
Should there be some error if the requested conversion isn't supported? I'm guessing people need the sampled vars downstream hence adding the node so is the failure mode more generally considered critical? Or, at least optionally so?

Reply:
What is the "requested conversion"? See my other comments about past usage of hard errors. But if you want to start a debate, note that I have subconsciously decided to go the other way for missing primvars in `computeSamplingFunction()`. But please don't start a debate :)

Reply:
Hehe, don't worry not going to start a debate. I guess its just down to what the use cases for the node are at the end of the day.... just thought I'd see if we had actual use cases in mind of it it was speculative either way.

Reply:
Perhaps I should rather say, please don't start a debate _now_. I do think there's stuff worth considering here, but what I've implemented is broadly in line with a convention which has grown out of experience, and I think any debate should consider the wider picture and its conclusions be applied everywhere. I don't this PR is the time for that.